src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -192,10 +192,6 @@ protected override async Task LoadComponentsAsync(CancellationToken cancellation
             // the appropriate task scheduler to report events on.
             this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
 
-            // Load and initialize the services detecting and adding new analyzer config documents as solution item.
-            this.ComponentModel.GetService<AnalyzerConfigDocumentAsSolutionItemHandler>().Initialize(this);
```

Comment:
Can we remove `AnalyzerConfigDocumentAsSolutionItemHandler` and `VisualStudioAddSolutionItemService` with this change?

Reply:
AnalyzerConfigDocumentAsSolutionItemHandler has been removed, but VisualStudioAddSolutionItemService  is used by ConfigurationUpdater

https://github.com/dotnet/roslyn/blob/76bdda7fe502a74f91aee0feb2fe251039be4e82/src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs#L288-L289

Reply:
It might be good to cleanup the interface, it looks like `IAddSolutionItemService.AddSolutionItemAsync` is only used by `AnalyzerConfigDocumentAsSolutionItemHandler` so we might be able to get rid of it.